ArduSub/joystick.cpp

```diff
@@ -849,6 +849,30 @@ void Sub::clear_input_hold()
     input_hold_engaged = false;
 }
 
+bool Sub::jsbutton_function_is_assigned(JSButton::button_function_t function)
+{
+    bool shift_is_assigned = false;
```

Comment:
It seems somewhat wasteful to determine this in here, rather than determining it outside (ideally just once), and passing it in to any and all button checks.

I don't think it's a big deal, but the same shift checking code runs redundantly >30 times from this PR alone.